### PR TITLE
Base functionality to serve a page

### PR DESCRIPTION
**Pull request by michaeldfallen**
This pull request contains the basic functionality ported from the Divorce frontend to serve a page.

I've copied only what's necessary to serve a basic html page, so there are a lot of things that are bare here, no real session stuff, no validation, no GOV.UK template or styling etc.

### CODE REVIEW

File: src/steps/Page.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const BaseStep = require('./BaseStep');
+const {expectImplemented} = require('../errors/expectImplemented');
```
Comment (paulgain) [inline src/steps/Page.js]: Currently this is not used, will you be using it going forward?

File: test/steps/Page.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+const Page = require('./../../src/steps/Page');
+const {request, testApp, testStep} = require('../util/supertest');
```
Comment (paulgain) [inline test/steps/Page.test.js]: request and testApp are required but not used.

File: test/util/supertest.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+const express = require('express');
+const supertest = require('supertest');
+const sessions = require('./../../src/services/sessions');
+const bodyParser = require('body-parser');
```
Comment (paulgain) [inline test/util/supertest.js]: bodyParser not used, I'm sure it will be in the future ;-)

**Comment by paulgain** [comment]
Is it possible we could restructure the steps so we have a directory structure "steps > hello-world" and inside hello-world we would have the HTML template and the step together, later we would add content and validation to the same directory, similar to what we currently have in Divorce.

File: src/Journey.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const express = require('express');
+const nunjucks = require('express-nunjucks');
+
```
Comment (paulgain) [inline src/Journey.js]: I'm not sure if coupling a specific version of express to the framework is ideal. What if a developer wants to use a specific version of express? However, we don't want to support every version of express. Thoughts?

File: src/steps/Page.js (hunk 1 above)
Comment (michaeldfallen) [inline src/steps/Page.js]: Ah, good point, a left over from a previous usage. I'll remove it.

File: test/steps/Page.test.js (hunk 2 above)
Comment (michaeldfallen) [inline test/steps/Page.test.js]: Dead imports from previous usage. :+1: Will remove.

File: test/util/supertest.js (hunk 3 above)
Comment (michaeldfallen) [inline test/util/supertest.js]: Damn dead imports, yep :+1: removing this.

File: src/Journey.js (hunk 4 above)
Comment (michaeldfallen) [inline src/Journey.js]: Yeah, originally I had wanted to this to be separated as well. The downside is that it means we can't handle the common express stuff as well, like template engine, serving assets, security stuff.

:thinking: Actually, looking back through the Divorce frontend it looks like the only parts that couldn't be encapsulated here would be:

```
    app.set('view engine', 'html');
    app.set('views', [
        __dirname + '/app/steps',
        __dirname + '/app/components',
        __dirname + '/app/views',
        __dirname + '/node_modules/govuk_template_jinja/views/layouts/'
    ]);

    nunjucks(app, {
        autoescape: true,
        watch: true,
        noCache: true
    });
```

so actually that's not so bad. Your app.js would be:
```
const app = express();
app.set('view engine', 'html');
app.set('views', [ ... ]);
nunjucks(app);

const journey = Journey(...)
...
```

**Comment by michaeldfallen** [comment]
@paulgain on the html in the same place as the step :+1: yep aiming for that. Haven't quite figured out how. In the old code we initialised all steps by doing `new Step('/path/to/step')` so that the step knew where to look for the template. I'm aiming to avoid that tight coupling. Something like having a templateFinder which generates a list of views folders for passing to `app.set('views', [...])` or having each step add it's folder to the views list.

Still thinking about that, that's the only reason I haven't got it here yet.

File: src/Journey.js (hunk 4 above)
Comment (paulgain) [inline src/Journey.js]: Yes, that looks far better. :+1:

**Comment by michaeldfallen** [comment]
@paulgain Ok, I've removed the express app part of Journey. I think you were right about that. 

I've also added linting (eslint) and cleaned every commit so those unused imports are removed now :+1:

**Comment by paulgain** [comment]
Good stuff, looking good.

**Pull request merged**